### PR TITLE
fix: Changes for Anvil Alpha release

### LINKED ISSUE #33718: [Bug]: Fixed layout widgets can be pasted on Anvil apps; albeit with compatibility issues
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Fixed layout widgets can be pasted on Anvil apps; albeit with compatibility issues

image

### Steps To Reproduce

1. Open a fixed layout app
2. Copy a few widgets
3. Open an Anvil app
4. Paste the widgets 

observe the widgets can be pasted but with errors in them

### Public Sample App

_No response_

### Environment

Deploy Preview

### Severity

High (Blocker to building or releasing)

### Issue video log

_No response_

### Version

cloud

### LINKED ISSUE #34575: [Task]: Create toggle for switching Anvil and Fixed modes in Applications page header
A toggle must be available for users to quickly switch between Anvil and Fixed mode development. 
This toggle will be available for all users for whom the toggle feature flag is enabled.
This toggle will appear in the header of the Applications page.

### LINKED ISSUE #34576: [Task]: Separate New and Classic applications in a workspace
### SubTasks

- [ ] If the Anvil toggle is enabled, users should see two sections within the workspace. 
- [ ] The New, Anvil, applications shall be listed first and the Classic Fixed mode applications will be listed second
- [ ] All relevant copies in the empty state or otherwise needs to be updated to properly represent the state of the system.
- [ ] Show Anvil alpha badge on the “New Applications” heading to represent the apps being Anvil applications.

### LINKED ISSUE #34578: [Task]: Update copy for Anvil alpha release
### SubTasks

- The paste mechanism is used by the cut-copy-paste workflow in Appsmith, as well as in the partial import of a page in Appsmith.
- A toast message is displayed to users when a paste or import is prohibited between two conflicting layout systems. 
- [ ] Show appropriate message to the user to help them understand the reason for preventing a paste of widgets.
- [ ] Show appropriate message when import app from git doesn’t show when Anvil is enabled.

### PR DESCRIPTION
**Pull request by riodeuno**
## Description
This PR makes the following changes to prepare for the Anvil Alpha release
- Add a toggle to the applications page header that allows us to toggle the Anvil feature flag from the UI.
- Enable the above toggle if the toggle feature flag is enabled
- When the toggle is enabled the following changes occur in the UX
  - Import application via Git is disabled and a callout is shown to the user
  - Git features are unavailable for users in Anvil applications
  - Copying and pasting widgets between two different layout systems is disabled and a warning message is shown to the user
  - Partial import export of widgets between two different layout systems is disabled and a warning message is shown to the user
  - CRUD page generation is disabled from datasources
  - Anvil applications and classic applications are separated into their own sections in the workspace.
  - In case any of the above sections are empty, an appropriate message is displayed in the empty section

### Screenshots

Alpha Toggle
Import from Git Callout
Paste-callout
Screenshot from 2024-07-03 14-39-13

### Other details  
- Toggle feature flag - `release_anvil_toggle_enabled`
- Anvil feature flag - `release_anvil_enabled`

Fixes #34578 
Fixes #34576 
Fixes #34575 
Fixes #33718

## Automation

/ok-to-test tags="@tag.All"

### :mag: Cypress test results

> [!CAUTION]
> 🔴 🔴 🔴 Some tests have failed.
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/9793128338>
> Commit: 09401272c1d52c915aee3b68925406f2f5f962ba
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=9793128338&attempt=2&selectiontype=test&testsstatus=failed&specsstatus=fail" target="_blank">Cypress dashboard</a>.
> Tags: @tag.All
> The following are new failures, please fix them before merging the PR: <ol>
> <li>cypress/e2e/Regression/ClientSide/PartialImportExport/PartialExport_spec.ts</ol>
> <a href="https://internal.appsmith.com/app/cypress-dashboard/identified-flaky-tests-65890b3c81d7400d08fa9ee3?branch=master" target="_blank">List of identified flaky tests</a>.

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
  - Added Anvil toggle functionality to the page header.
  - Introduced layout compatibility checks for pasting widgets.

- **Enhancements**
  - Updated widget paste action to verify layout system compatibility.
  - Improved handling and processing of user-uploaded JSON files for widget import.

### CODE REVIEW

**Review by marks0351 — APPROVED** [review]

**Review by KelvinOm — APPROVED** [review]

File: app/client/src/pages/common/PageHeader.tsx [hunk 1]
```diff
@@ -61,6 +81,26 @@ export function PageHeader(props: PageHeaderProps) {
   const showBanner = useSelector(shouldShowLicenseBanner);
   const isHomePage = useRouteMatch("/applications")?.isExact;
   const isLicensePage = useRouteMatch("/license")?.isExact;
+  const isAnvilEnabled = useSelector(getIsAnvilLayoutEnabled);
+  const shouldShowAnvilToggle = useFeatureFlag(
+    FEATURE_FLAG.release_anvil_toggle_enabled,
+  );
+
+  log.debug("Is Anvil Enabled:", isAnvilEnabled);
```
[the remark this answers is not in the document]
Comment (riodeuno) [reply app/client/src/pages/common/PageHeader.tsx:89]: These are useful when debugging on user's machines. If you set the LOGLEVEL  to DEBUG in user's machines, you can see these logs.

[the remark this answers is not in the document]
Comment (riodeuno) [reply app/client/src/pages/common/PageHeader.tsx:89]: I will remove this next week, when I'm sure there are no zero day bugs.

[the remark this answers is not in the document]
Comment (riodeuno) [reply app/client/src/pages/common/PageHeader.tsx]: @ichik pointed out an issue, that was helped by this log. Now that I've verified that this is working, multiple times. I have removed the logs.

**Pull request merged**